benchmark/training/training_benchmark.py

```diff
@@ -0,0 +1,189 @@
+                                print(f'Calculated degree for {dataset_name}.')
+                            params['degree'] = degree
+
+                        model = get_model(
+                            model_name, params,
+                            metadata=data.metadata() if hetero else None)
+                        model = model.to(device)
+                        model.train()
+                        optimizer = torch.optim.Adam(model.parameters(),
+                                                     lr=0.001)
+
+                        progress_bar = False if args.no_progress_bar else True
+                        with amp:
+                            for _ in range(args.warmup):
+                                train(model, subgraph_loader, optimizer,
+                                      device, hetero,
+                                      progress_bar=progress_bar, desc="Warmup")
+                            with timeit():
+                                for epoch in range(args.num_epochs):
+                                    train(model, subgraph_loader, optimizer,
+                                          device, hetero,
+                                          progress_bar=progress_bar,
+                                          desc=f"Epoch={epoch}")
+
+                            if args.profile:
+                                with torch_profile():
+                                    for epoch in range(args.num_epochs):
+                                        train(model, subgraph_loader,
+                                              optimizer, device, hetero,
+                                              progress_bar=progress_bar,
+                                              desc=f"Epoch={epoch}")
+                                rename_profile_file(model_name, dataset_name,
+                                                    str(batch_size),
+                                                    str(layers),
+                                                    str(hidden_channels),
+                                                    str(num_neighbors))
+
+
+if __name__ == '__main__':
+    argparser = argparse.ArgumentParser('GNN training benchmark')
+    argparser.add_argument('--datasets', nargs='+',
+                           default=['ogbn-mag', 'ogbn-products',
+                                    'Reddit'], type=str)
+    argparser.add_argument(
+        '--use-sparse-tensor', action='store_true',
+        help='use torch_sparse.SparseTensor as graph storage format')
+    argparser.add_argument(
+        '--models', nargs='+',
+        default=['edge_cnn', 'gat', 'gcn', 'pna', 'rgat', 'rgcn'], type=str)
+    argparser.add_argument('--root', default='../../data', type=str,
+                           help='relative path to look for the datasets')
+    argparser.add_argument('--batch-sizes', nargs='+',
+                           default=[512, 1024, 2048, 4096, 8192], type=int)
```

Comment:
For benchmark purpose, we need to guarantee benchmark can be finished in several minutes. Otherwise, it is not a good benchmark. Have you test all the options with `batch-sizes`, `num-layers`, `num-hidden_chennels`? And how much memory need to use at the most? Is the memory consumption Okay for user?
This may be also an issue in benchmark/inference. Just highlight here. We can adjust the options offline to fit the benchmark requirement.